app/Http/Middleware/LogAction.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+
+class LogAction
+{
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (Auth::check() && Auth::user()->isAdmin() && $request->method() !== 'GET') {
+            parse_str($request->getContent(), $body);
+            $bodyKeys = [];
+            foreach ($body as $key => $value) {
+                array_push($bodyKeys, $key); // Only include keys from request body to avoid logging any sensitive values.
+            }
```

Comment:
$request->getContent() seems to return just a single string, which gets logged as one giant key. Instead of getContent, try using $request->input(), which matches how we access request data in our controllers.